### ISSUE TITLE
rename app distribution APK on upload

### ISSUE THREAD

**Issue by michaelaubertcmc** [firebase/firebase-android-sdk#5044]
## What feature would you like to see?

The app distribution gradle plugin should let testers download APKs with the same file name as when they were uploaded

Testers can currently only download files named "app.apk" no matter the name of the apk file that was originally uploaded

Using the "artifactPath" parameter in gradle is a problem when the Android application has more than one build variant.

Could we add a "artifactName"?

## How would you use it?

This would let us keep meaningful names of APKs when distributing them from developers to testers. It would allow us to avoid mistakes when testing our mobile application

Both Jenkins and Gitlab let us control the name of the artifacts. We would like Firebase App Distribution to be our source of truth instead.

**Comment by harishrpatel** [comment]
Yes, would love this feature for build variants/flavors. We use Github Actions and it would be great if we could pass build number like ${{ github.run_number }} and have it be part of the file name (if possible).

**Comment by tandrej98** [comment]
Hi, is there any update on this please?

**Comment by SiddheshShetye** [comment]
This is really needed feature. any update on this?

**Comment by argzdev** [comment]
Hi folks, thanks for pinging this to our attention. Let me try bringing this up to our engineers to get some feedback.

**Comment by jpgpuyo** [comment]
Any updates on this? Thanks!

**Comment by tandrej98** [comment]
Hi there, any updates here please?

**Comment by masztalski** [comment]
Just found out about this issue as freshly moved from appCenter to firebase distribution. File should stay at least exactly same as it was uploaded. Sure, some extra control with parameters will be great, but at least stop renaming it

**Comment by SiddheshShetye** [comment]
This thing is very annoying and taking away us from using app distribution. update on this will be highly appreciated.

**Comment by rexmtorres** [comment]
Just now, a tester from one of our clients complained that when they installed our app, it replaced one of their other apps.  It turned out they had another "app.apk" file for another app in their download folder and installed that instead of our app.  So, I would really appreciate this feature to be able to rename the APK.

**Comment by petered** [comment]
Please Google!  This seems easy to solve and high-impact.  It's almost always the case when testing new apps that you have a bunch of versions that you download, and having them called app(1).apk, app(2).apk, .... etc is really unhelpful when you're trying to try out different versions.

**Comment by codenamesid** [comment]
One more Vote to please google for this

**Comment by bhallionOhbibi** [comment]
Please, make this happen

**Comment by khushank** [comment]
This is really requires as it creates confusions to developers as to what version to download.

**Comment by EmilADamian** [comment]
Yes, please do something about it.

Thanks

**Comment by OBCZ** [comment]
Another soul migrating from AppCentrer to Firebase App Distribution. This feature would be very helpful indeed. 🙏 Thanks in advance!

**Comment by sbcastle** [comment]
Yet another migration from AppCenter to Firebase and found this issue.   ComeON!

**Comment by mhein-mm** [comment]
Any updates?

**Comment by blackwiz4rd** [comment]
Updates? This is very frustating for me when sharing files to other people

**Comment by blackwiz4rd** [comment]
It's incredible that this issue is still open after several years, and it's yet something so simple to fix...

**Comment by petered** [comment]
"Claude, can you rebuild a software service that is identical to Firebase
except that in the app testing framework it allows you to customize the
name of your freaking APK? "

**Comment by mayo-tds** [comment]
hey!! this would be extremely useful for QA teams that have to download a bunch of builds to do QA passes over different projects and/or versions in a close timeframe!!!
